src/kOS.Safe/Encapsulation/KOSDelegate.cs

```diff
@@ -0,0 +1,154 @@
+        }
+        
+        public KOSDelegate(KOSDelegate oldCopy)
+        {
+            cpu = oldCopy.cpu;
+            InitializeSuffixes();
+            foreach (object ca in oldCopy.preBoundArgs)
+                preBoundArgs.Add(ca);
+        }
+
+        private void InitializeSuffixes()
+        {
+            AddSuffix("CALL", new VarArgsSuffix<object, object>(Call));
+            AddSuffix("BIND", new VarArgsSuffix<KOSDelegate, object>(Bind));
+        }
+
+        public void AddPreBoundArg(object arg)
+        {
+            preBoundArgs.Add(arg);
+        }
+
+        public object Call(params object[] args)
+        {
+            PushUnderArgs();
+            cpu.PushStack(new KOSArgMarkerType());
+            foreach (object arg in preBoundArgs)
+            {
+                cpu.PushStack(arg);
+            }
+            foreach (object arg in args)
+            {
+                cpu.PushStack(arg);
+            }
+            return Call();
+        }
+        
+        /// <summary>
+        /// Assuming normal args are already on the stack,
+        /// then the prebound args would have to be inserted
+        /// underneath them, which slightly violates the
+        /// stack access rules.  So do it through this method
+        /// if it's needed.  (If Call() was used, then it's not needed
+        /// because Call() does this for you.  But if Call() was
+        /// not used and someone just used bare parentheses like so:
+        ///    function y { parameter a,b,c. print a+b+c. }
+        ///    set x to y@:bind(1,2).
+        ///    x(). // instead of saying x:call().
+        /// then the prebound args don't get pushed by doing that, and they
+        /// have to get pushed under the top by calling this.)
+        /// </summary>
+        public void InsertPreBoundArgs()
+        {
+            Stack<object> aboveArgs = new Stack<object>();
+            object arg = ""; // doesn't matter what it is as long as it's non-null for the while check below.
+            while (arg != null && !(arg is KOSArgMarkerType))
+            {
+                arg = cpu.PopStack();
+                if (! (arg is KOSArgMarkerType))
+                    aboveArgs.Push(arg);
+            }
+            if (arg == null)
+                throw new KOSException("KOSDelegate.InsertPreBoundArgs: Stack arg bottom missing.\n" +
+                                       "Contact the kOS devs.  This message should 'never' happen.");
+            // Now re-push the args back, putting the preBound ones at the bottom
+            // where they belong:
+            cpu.PushStack(new KOSArgMarkerType());
+            foreach (object item in preBoundArgs)
+            {
+                cpu.PushStack(item);
+            }
+            foreach (object item in aboveArgs) // Because this was pushed to a stack, this should show in reverse order.
+            {
+                cpu.PushStack(item);
+            }
+        }
```

Comment:
It occurs to me that PushUnderArgs() is only being called from Call(), not from here.  That may explain why I'm having some troubles with cases where I don't explictitly state the ``:call`` suffix.  I'll have to double-check this tomorrow.